src/github.com/matrix-org/dendrite/roomserver/types/types.go

```diff
@@ -75,3 +83,15 @@ const (
 	// EmptyStateKeyNID is the numeric ID for the empty state key.
 	EmptyStateKeyNID = 1
 )
+
+// StateDataNIDList is used to return the result of bulk StateDataNID lookups from the database.
+type StateDataNIDList struct {
```

Comment:
It feels like we do this a lot: define a struct simply to represent a table schema. I think we should be more upfront with this in the type names, so people don't think there is a semantic Matrixy meaning to these structs. Some convention like appending `Result` (as in, Result Set) may be an idea? E.g `StateDataNIDListResult`

WDYT?